JumpscaleCore/data/schema/SchemaFactory.py

```diff
@@ -175,32 +189,30 @@ def _schema_text_rewrite(self, url, schema_text):
         """
         schema_text = j.core.tools.text_strip(schema_text)
         found_nrs = False
-        nr = 0
+        nr_generated = 0
         out = ""
+
         for line in schema_text.split("\n"):
             line = line.strip()
+            line_nr = self._get_line_number(line)
+            found_nrs |= bool(line_nr)
+
             line = line.replace("::", ":")  # there was some but at one point of time
             if line.startswith("@url"):
                 if url:
                     continue
                 else:
                     url = line.split("=", 1)[1].strip()
                     continue
-            if line.startswith("@"):
+            if line.startswith("@") or line.startswith("#") or line_nr:
```

Comment:
not sure this is correct, everything has a line nr, think logic broken

Reply:
the check for found nr's is gone

Reply:
Maybe i'm wrong, but what's the difference between this:

```python
            if line.startswith("@") or line.startswith("#") or line_nr:
                out += "%s\n" % line
```

and this? (i understand it like this condition checks for line number? `elif ":" in line:`, right?)

```python
            if line.startswith("@"):
                out += "%s\n" % line
            elif line.strip() == "":
                out += "\n"
            elif line.strip().startswith("#"):
                out += "%s\n" % line
            elif ":" in line:
                found_nrs = True
                out += "%s\n" % line
```

Also, found_nrs is set to true if line_nr is true, using the compound operator here:

```python
            found_nrs |= bool(line_nr)
```